test/base.py

```diff
@@ -229,7 +232,7 @@ def is_ssl_handshake_error(self, exit_status):
                           RetryableTapError,
                           max_tries=2,
                           factor=30)
-    def run_and_verify_sync(self, conn_id):
+    def run_and_verify_sync(self, conn_id, state=None):
```

Comment:
should we default state to `{}` so that this always resets state to a fresh connection unless specified?

Reply:
Actually, doing so could lead to a very nasty bug 
https://florimond.dev/en/posts/2018/08/python-mutable-defaults-are-the-source-of-all-evil/

Reply:
TLDR
> In Python, when passing a mutable value as a default argument in a function, the default argument is mutated anytime that value is mutated.

The blog post uses a list as the example, but you get the idea

Reply:
I'm not sure about the convention in general but in the zendesk repo we seem to set state to {} or "state" in the tests to clear it or update it.  Could it help from a debug perspective to have base set it to None so we can tell if the test has set state somewhere or if it's never been set?  What's the argument for always setting it to {}?  I'm not sure I have a strong opinion either way.